ci-automation/vendor-testing/azure.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+# Copyright (c) 2021 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -euo pipefail
+
+set -x
+
+# Test execution script for the qemu vendor image.
+# This script is supposed to run in the SDK container.
+
+# $@ now contains tests / test patterns to run
+
+source ci-automation/vendor_test.sh
+
+board="${CIA_ARCH}-usr"
+basename="ci-${CIA_VERNUM//+/-}"
+azure_instance_type="${AZURE_MACHINE_SIZE}"
+azure_vnet_subnet_name="jenkins-vnet-${AZURE_LOCATION}"
+
+azure_profile_config_file=''
+secret_to_file azure_profile_config_file "${AZURE_PROFILE}"
+azure_auth_config_file=''
+secret_to_file azure_auth_config_file "${AZURE_AUTH_CREDENTIALS}"
+
+testscript="$(basename "$0")"
+
+# Fetch the Azure image if not present
+if [ -f "${AZURE_IMAGE_NAME}" ] ; then
+    echo "++++ ${testscript}: Using existing ${work_dir}/${AZURE_IMAGE_NAME} for testing ${CIA_VERNUM} (${CIA_ARCH}) ++++"
+else
+    echo "++++ ${testscript}: downloading ${AZURE_IMAGE_NAME} for ${CIA_VERNUM} (${CIA_ARCH}) ++++"
+    copy_from_buildcache "images/${CIA_ARCH}/${CIA_VERNUM}/${AZURE_IMAGE_NAME}.bz2" .
+    cp --sparse=always <(lbzcat "${AZURE_IMAGE_NAME}.bz2") "${AZURE_IMAGE_NAME}"
+    rm "${AZURE_IMAGE_NAME}.bz2"
+fi
+
+if [[ "${CIA_ARCH}" == "arm64" ]]; then
+  AZURE_USE_GALLERY="--azure-use-gallery"
+fi
+
+set -o noglob
+
+run_kola_tests() {
+    local instance_type="${1}"; shift
+    local instance_tapfile="${1}"; shift
+
+    timeout --signal=SIGQUIT 20h \
+    kola run \
+    --board="${board}" \
+    --basename="${basename}" \
+    --parallel="${AZURE_PARALLEL}" \
+    --offering=basic \
+    --platform=azure \
+    --azure-image-file="${AZURE_IMAGE_NAME}" \
+    --azure-location="${AZURE_LOCATION}" \
+    --azure-profile="${azure_profile_config_file}" \
+    --azure-auth="${azure_auth_config_file}" \
+    --torcx-manifest=${CIA_TORCX_MANIFEST} \
+    --tapfile="${instance_tapfile}" \
+    --azure-size=${instance_type} \
+    ${AZURE_USE_GALLERY} \
+    ${AZURE_HYPER_V_GENERATION:+--azure-hyper-v-generation=${AZURE_HYPER_V_GENERATION}} \
+    ${azure_vnet_subnet_name:+--azure-vnet-subnet-name=${azure_vnet_subnet_name}} \
+    ${AZURE_USE_PRIVATE_IPS:+--azure-use-private-ips=${AZURE_USE_PRIVATE_IPS}} \
+    "${@}"
+}
+
+query_kola_tests() {
+    shift; # ignore the instance type
+    kola list --platform=azure --filter "${@}"
+}
+
+run_kola_tests_on_instances \
+    "${azure_instance_type}" \
+    "${CIA_TAPFILE}" \
+    "${CIA_FIRST_RUN}" \
+    "" \
```

Comment:
For an MVP we can ignore running for multiple instance types but let's focus on testing both v1 and v2 at least.
```suggestion
    ${IS_AMD64:+v1} \
```
I would assume it makes sense to test on v2 by default and just run the cl.internet test on v1?